packages/nimble-components/src/wafer-map/modules/experimental/hover-handler.ts

```diff
@@ -0,0 +1,99 @@
+import type { WaferMap } from '../..';
+import { PointCoordinates, WaferMapOriginLocation } from '../../types';
+
+/**
+ * HoverHandler deals with user interactions and events like hovering
+ */
+export class HoverHandler {
+    public constructor(private readonly wafermap: WaferMap) {}
+
+    /**
+     * @internal
+     */
+    public connect(): void {
+        this.wafermap.addEventListener('mousemove', this.onMouseMove);
+        this.wafermap.addEventListener('mouseout', this.onMouseOut);
+    }
+
+    /**
+     * @internal
+     */
+    public disconnect(): void {
+        this.wafermap.removeEventListener('mousemove', this.onMouseMove);
+        this.wafermap.removeEventListener('mouseout', this.onMouseOut);
+    }
+
+    /**
+     * @internal
+     * keep public for testing until data manager refactor
+     */
+    public readonly onMouseMove = (event: MouseEvent): void => {
+        if (!this.wafermap.isExperimentalRenderer()) {
+            return;
+        }
+        // get original mouse position in case we are in zoom.
+        const invertedPoint = this.wafermap.transform.invert([
+            event.offsetX,
+            event.offsetY
+        ]);
+
+        // does not work yet until data manager will parse diesTable
+        const dieCoordinates = this.calculateDieCoordinates(this.wafermap, {
+            x: invertedPoint[0],
+            y: invertedPoint[1]
+        });
+        const colIndex = this.wafermap
+            .diesTable!.getChild('colIndex')!
+            .toArray() as Int32Array;
+        const rowIndex = this.wafermap
+            .diesTable!.getChild('rowIndex')!
+            .toArray() as Int32Array;
```

Comment:
We should not need all this `as` casting, we already reviewed how we can type `Table<>` correctly. You can see in this example how TypeScript will know the types if we type it correctly: https://stackblitz.com/edit/arrow-table-generics?file=src%2Fmain.ts

![image](https://github.com/ni/nimble/assets/1588923/f59325c7-14a6-47e9-ac25-2e3933a7eff6)

Reply:
I made the generic table but now the precision and bitWidth validity checks give type erros. should we keep them? Is there a way to make thwm work?

Reply:
Conceptually yes. These are web components so a user could leverage them in a JS application without type checking in which case we need the validator for runtime type checking. All the typescript work helps TypeScript users and us as maintainers to find errors at compile time via static analysis.

Looking at the CI build errors my guess is it is related to the extra fields added to the type:
```
export interface WaferRequiredTypeMap {
    [key: string]: DataType;
    [key: number]: DataType;
    [key: symbol]: DataType;
    colIndex: Int32;
    rowIndex: Int32;
    value: Float64;
}
```
ie the:
```
    [key: string]: DataType;
    [key: number]: DataType;
    [key: symbol]: DataType;
```

Either way I'd suggest we not add those yet, there are still concerns about that API that are not resolved and it does not seem necessary to include those yet.

Reply:
removing them leaves this error: Type 'T' does not satisfy the constraint 'TypeMap'.
  Type 'WaferRequiredTypeMap' is not assignable to type 'TypeMap'.